src/API/Management/Users.php

```diff
@@ -1,88 +1,262 @@
+        return $this->apiClient->method('get')
+            ->addPath('users')
+            ->withDictParams($params)
+            ->call();
     }
 
-    public function delete($user_id) {
+    /**
+     * Wrapper for self::search().
+     *
+     * @param array $params - Search parameters to send.
+     *
+     * @return mixed|string
+     *
+     * @throws \Exception
+     *
+     * @see self::getAll()
+     */
+    public function search($params = [])
+    {
+        return $this->getAll($params);
+    }
 
-        return $this->apiClient->delete()
-            ->users($user_id)
+    /**
+     * Delete a User by ID.
+     *
+     * @param string $user_id - User ID to delete.
+     *
+     * @return mixed|string
+     *
+     * @throws \Exception
+     */
+    public function delete($user_id)
+    {
+        return $this->apiClient->method('delete')
+            ->addPath('users', $user_id)
             ->call();
     }
 
-    public function linkAccount($user_id, $post_identities_body) {
-
-        return $this->apiClient->post()
-            ->users($user_id)
-            ->identities()
-            ->withHeader(new ContentType('application/json'))
-            ->withBody(json_encode($post_identities_body))
+    /**
+     * Link one user account to another.
+     *
+     * @param string $user_id - User ID of the primary identity where you are linking the secondary account to.
+     * @param array $data - Secondary account to link; either link_with JWT or provider, connection_id, and user_id.
+     *
+     * @return array - Array of the primary account identities.
+     *
+     * @throws \Exception
+     */
+    public function linkAccount($user_id, $data)
+    {
+        return $this->apiClient->method('post')
+            ->addPath('users', $user_id)
+            ->addPath('identities')
+            ->withBody(json_encode($data))
             ->call();
     }
 
-    public function unlinkAccount($user_id, $provider, $identity_id) {
-
-        return $this->apiClient->delete()
-            ->users($user_id)
-            ->identities($provider)
+    /**
+     * Unlink an identity from the target user.
+     *
+     * @param string $user_id - User ID to unlink.
+     * @param string $provider - Identity provider of the secondary linked account.
+     * @param string $identity_id- The unique identifier of the secondary linked account.
+     *
+     * @return mixed|string
+     *
+     * @throws \Exception
+     */
+    public function unlinkAccount($user_id, $provider, $identity_id)
+    {
+        return $this->apiClient->method('delete')
+            ->addPath('users', $user_id)
+            ->addPath('identities', $provider)
             ->addPathVariable($identity_id)
             ->call();
     }
 
-    public function unlinkDevice($user_id, $device_id) {
-        return $this->apiClient->delete()
-            ->users($user_id)
-            ->devices($device_id)
+    /**
+     * TODO: Deprecate, endpoint does not exist.
```

Comment:
if the endpoint doesn't exist at all, what's the point on keeping this as "deprecated"? In that case, the right action would be removing it. (On separate PR)

Reply:
The difference is that a missing method throws a fatal error whereas a 404 from the HTTP client throughs a specific exception.

Reply:
You have a point. What about just throwing that specific exception manually instead of calling the endpoint? I don't see the point on making an actual request if I know in advance it's going to fail. At least, log something before making the call for the dev to know what to use now.

Reply:
I didn't see a suitable replacement ... and I'm not totally sure what this should be doing in the first place.